tests/test_categories.py

```diff
@@ -0,0 +1,147 @@
+  children:
+  - name: pressure
+    children:
+    - name: absolute pressure
+    - name: relative pressure
+    - name: pressure difference
+  - name: impedance
+    children:
+    - name: absolute impedance
+    - name: relative impedance
+    - name: impedance difference
+"""
+
+categories_dict = {
+    "name": "category",
+    "children": [
+        {
+            "name": "physical measurements",
+            "children": [
+                {
+                    "name": "pressure",
+                    "children": [
+                        {"name": "absolute pressure"},
+                        {"name": "relative pressure"},
+                        {"name": "pressure difference"},
+                    ],
+                },
+                {
+                    "name": "impedance",
+                    "children": [
+                        {"name": "absolute impedance"},
+                        {"name": "relative impedance"},
+                        {"name": "impedance difference"},
+                    ],
+                },
+            ],
+        },
+    ],
+}
+
+
+@pytest.fixture()
+def categories_from_dict() -> Category:
+    return Category.from_dict(categories_dict)
+
+
+def test_no_empty_init():
+    with pytest.raises(TypeError):
+        _ = Category()
+
+
+def test_load_from_dict():
+    _ = Category.from_dict(categories_dict)
+
+
+def test_load_from_yaml():
+    _ = Category.from_yaml(yaml_string)
+
+
+def test_get_item(categories_from_dict: Category):
+    node = categories_from_dict["pressure difference"]
+    assert isinstance(node, Category)
+    assert node.name == "pressure difference"
+
+
+def test_has_child(categories_from_dict: Category):
+    assert categories_from_dict.has_subcategory("pressure difference")
+    assert categories_from_dict["pressure"].has_subcategory("pressure difference")
+    assert not categories_from_dict["pressure"].has_subcategory("impedance difference")
+
+    with pytest.raises(ValueError):
+        categories_from_dict["foo"]
+
+
+def test_get_default_categories():
+    categories = get_default_categories()
+    assert isinstance(categories, Category)
+    assert categories.has_subcategory("pressure")
+    assert categories.has_subcategory("other")
+
+    second_time = get_default_categories()
+    assert second_time is categories
+
+
+def test_get_multiple(categories_from_dict: Category):
+    subset = categories_from_dict[["absolute pressure", "absolute impedance"]]
+    assert len(subset.descendants) == 2
+    assert {d.name for d in subset.descendants} == {"absolute pressure", "absolute impedance"}
+    assert subset.has_subcategory("absolute pressure")
+    assert categories_from_dict.has_subcategory("absolute pressure")
+    assert categories_from_dict.has_subcategory("absolute impedance")
+    assert subset["absolute pressure"] is not categories_from_dict["absolute pressure"]
+    assert subset["absolute impedance"] is not categories_from_dict["absolute impedance"]
```

Comment:
For my understanding: are these not the same because when selecting multiple subcategories you create a new tree with a temporary root and thus you get different objects for subset["absolute pressure"] and categories_from_dict["absolute pressure"]?

Reply:
Yes, indeed. A category can only be part of one tree. Here we make a new tree with a temporary root. The categories in that temporary tree are copies, not the original objects.